sdcm/teardown_validators/rackaware.py

```diff
@@ -0,0 +1,167 @@
+import logging
+import time
+
+from sdcm.sct_events import Severity
+from sdcm.sct_events.teardown_validators import ValidatorEvent
+from sdcm.teardown_validators.base import TeardownValidator
+
+LOGGER = logging.getLogger(__name__)
+
+
+class RackawareValidator(TeardownValidator):  # pylint: disable=too-few-public-methods
+    """
+    Rack-aware validation is enabled under these conditions:
+    - A rack-aware policy is configured.
+    - A single loader is employed (due to Scylla rack metric limitation).
+    - The cluster is deployed across multiple availability zones or a simulated rack environment or in multiple regions.
+
+    The validation aims to confirm that no coordination traffic is directed to availability zones where loaders are absent
+    """
+    validator_name = 'rackaware'
+
+    # This variable defines the allowed percentage variation of system CQL reads directed to availability zones without loaders.
+    # These reads, such as those performed by nemeses or non-rack-aware test code, may be routed this way.
+    EXPECTED_NON_SYSTEM_READS_PERC = 10
+
+    @property
+    def count_loaders(self):
+        loaders = sum(len(loaders.nodes) for loaders in self.tester.loaders_multitenant)
```

Comment:
IIRC `loaders_multitenant` and `db_clusters_multitenant` refer to multitenancy, not multi dc
I think you need to change if cases - also we might have 2 loaders in 1st dc and 0 in others.

Reply:
for non multitenancy cases `loaders_multitenant == loaders`

for multitenancy, this computation should be per tenant, but for now it would be o.k. not to support multitenancy.

Reply:
In a multitenant setup where each tenant has its own loader(s), validation is confined to individual tenants, disregarding loaders belonging to others. 
Despite the multitenant context, the `count_loaders` function operates at a cluster level, effectively checking if the entire cluster (regardless of tenant separation) has only one loader.

Reply:
> In a multitenant setup where each tenant has its own loader(s), validation is confined to individual tenants, disregarding loaders belonging to others.

but you use `sum` - it looks like loaders from other tenants are counted

> Despite the multitenant context, the `count_loaders` function operates at a cluster level, effectively checking how many loaders the entire cluster (regardless of tenant separation) has. This amount will be used to understand if the cluster has one ore more loaders

in multidc when each dc has 1 loader, shouldn't it be valid case for verification of `rack-aware policy`?

Reply:
removed multitenance using